app/[lang]/archive/page.tsx

```
@@ -1,22 +1,174 @@
+    },
+    paddingTop: {
+      xs: '80px',
+      sm: '89px',
+      md: '89px',
+      lg: '95px',
+      xl: '97px'
+    }
+  },
+  fundsGrid: {
+    gridColumn: '1 / -1',
+    display: 'grid',
+    gridTemplateColumns: {
+      xs: 'repeat(1, 1fr)',
+      sm: 'repeat(2, 1fr)',
+      md: 'repeat(3, 1fr)',
+      lg: 'repeat(4, 1fr)'
+    },
+    columnGap: {
+      xs: '16px',
+      sm: '24px',
+      md: '32px',
+      lg: '40px'
+    },
+    paddingBottom: {
+      xs: '80px',
+      sm: '96px',
+      md: '80px',
+      lg: '96px'
+    }
+  }
+};
+
+const getColumnPaddingTop = (columnNum: number) => {
+  const paddingMap = {
+    sm: columnNum === 1 ? '40px' : '0px',
+    md: (() => {
+      if (columnNum === 1) return '80px';
+      if (columnNum === 2) return '40px';
+      return '0px';
+    })(),
+    lg: (() => {
+      if (columnNum === 1) return '120px';
+      if (columnNum === 2) return '80px';
+      if (columnNum === 3) return '40px';
+      return '0px';
+    })()
+  };
+
+  return {
+    xs: '0px',
+    ...paddingMap
+  };
+};
+
+export default function Archive() {
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+  const isSmallTablet = useMediaQuery(theme.breakpoints.between('sm', 'md'));
+  const isBigTablet = useMediaQuery(theme.breakpoints.between('md', 'lg'));
+  const [error, setError] = useState<string | null>(null);
+
+  const [funds, setFunds] = useState<FundDTO[]>([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const loadFunds = async () => {
+      try {
+        const data = await newFundsRepository().getFunds();
+        setFunds(data);
+      } catch (err) {
+        setError(err instanceof Error ? err.message : 'Unknown error');
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    loadFunds();
+  }, []);
+
+  if (error) {
+    return <div>Error: {error}</div>;
+  }
```

you can throw an error there is a error boundary on root layout that will catch it and display Error page